doc/asciidoc/algorithms/alpha/alpha-particle-filtering.adoc

```diff
@@ -0,0 +1,99 @@
+* <<algorithms-particle-filtering-syntax, Syntax>>
+* <<algorithms-particle-filtering-examples, Examples>>
+
+[[algorithms-particle-filtering-context]]
+== History and explanation
+
+Particle Filtering was proposed by Denis Gallo, Matteo Lissandrini, and Yannis Velegrakis in their 2020 paper https://people.cs.aau.dk/~matteo/pdf/EDBT20-particle-filtering.pdf[Personalized Page Rank on Knowledge Graphs: Particle Filtering is all you need!^].
+
+[[algorithms-particle-filtering-usecase]]
+== Use-cases - when to use the Particle Filtering algorithm
+
+* Particle filtering has been used to build the movie recommendation engine used by https://mindreader.tech/[MindReader^], a novel dataset providing explicit user ratings over a knowledge graph within the movie domain.
+Read more in https://towardsdatascience.com/movie-recommendations-powered-by-knowledge-graphs-and-neo4j-33603a212ad0[Movie Recommendations powered by Knowledge Graphs and Neo4j^].
+
+[[algorithms-particle-filtering-syntax]]
+== Syntax
+
+.The following will run the algorithm and stream back results:
+[source, cypher]
+----
+CALL gds.alpha.particleFiltering.stream(configuration: Map)
+YIELD nodeId, score
+----
+
+.Configuration
+[opts="header",cols="1,1,1,1,4"]
+|===
+| Name                          | Type    | Default                | Optional | Description
+| sourceNodes                | List    | []        | yes      | A set of nodes to use for computing Particle Filtering.
+| numberParticles                   | int     | 10                      | yes      | The number of particles
+| minimumThreshold               | float     | 0.5 | yes      | The minimum threshold for scores. Values below this will not be returned.
+| dampingFactor                 | float   | 0.85                   | yes      | The damping factor of the Particle Filtering calculation.
+| relationshipType                 | String   | null                   | yes      | The relationship type to use.
+| direction                 | String   | OUTGOING                   | yes      | The direction of the relationship type.
+| shuffleNeighbors                 | Boolean   | true                   | yes      | Whether to shuffle neighboring nodes before iterating over them.
+|===
+
+[[algorithms-particle-filtering-examples]]
+== Examples
+
+In this section we will show examples of executing the Particle Filtering algorithm on a concrete graph.
+The intention is to illustrate what the results look like and to provide a guide in how to make use of the algorithm in a real setting.
+The example graph looks like this:
+
+image::page-rank-graph.svg[align="center"]
+
+.The following Cypher statement will create the example graph in the Neo4j database:
+[source, cypher, role=setup-query]
+----
+include::../page-rank/sample-graph.cypher[tag=setup]
+----
+
+The following examples show how to run Particle Filtering centered around 'Site A'.
+
+[role=query-example]
+--
+.The following will run the algorithm and stream results:
```

Comment:
It is important to explain the meaning of the parameters and how to select them.
In particular, the number of particles w.r.t. the minimum threshold.

Reply:
> * The documentation is missing the explanation for the parameters, these are fundamental to obtain good results (p.s. I can provide more text for the documentation about these)

Have updated the citations and added more documentation. Could you provide some more text for documenting the parameters? 

I'll add the labelled version of the algorithm as well. I just want to check that I understand how the weights are computed here: https://github.com/DenisGallo/Neo4j-ParticleFiltering/blob/master/src/main/java/pfiltering/WeightingGraphProc.java#L30

* Get total relationship count
* Get the count of each relationship type
* Store a score on each relationship of: `-1 * log(1.0*relTypeCount/totalRelCount)`

If that's the correct formula, I don't think we need to compute that in a separate procedure. Neo has internal stats stores that count the number of total relationships & the number of each type of relationship, so we can retrieve those inside the particleFiltering procedure call at low cost.

Reply:
**Parameters explanations:**
`numberParticles`:   controls the granularity of the computation, and consequently the precision of the final result. An ideal initial number of particles is some multiple of the degree of the source query  nodes.

`minThreshold`: minimum should be `1/numberParticles` . This determines the minimum amount of PPR score for  a node to be returned as result.  if set to `1/numberParticles` it should return all nodes visited. Higher value will return only nodes with some non trivial score.


`dampingFactor` : this controls how far from the source nodes the PPR algorithm should navigate. `0.85` usually provides a good balance. For a very sparse and large graph (in terms of diameter) values around `0.9-0.95` can be also used.